_posts/2023-03-06-NGINX-Dashboards-with-Fluent-Bit.md

```diff
@@ -0,0 +1,68 @@
+
+Simple Schema for Observability is inspired by both [OpenTelemetry](https://opentelemetry.io/docs/) and uses Amazon Elastic Container Service ([Amazon ECS](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/ecs_cwe_events.html)) event logs and OpenTelemetry metadata.
+
+## Connecting Simple Schema and Fluent Bit
+
+When raw logs are ingested by Fluent Bit they are automatically converted into MessagePack, a binary representation of JSON. Fluent Bit’s automatic conversion makes parsing and modifying messages in-flight simple and easy.
+
+In addition, Fluent Bit comes with out of the box parsers for common applications such as Nginx, Apache Web Logs, Kubernetes, and more. These parsers give structure to a log file, though they do not have the depth that other log schemas and formats do. Thankfully, Fluent Bit can also handle more complex transformations along with with processors (also known as filters), or even Lua code.
+
+[Lua filters](https://docs.fluentbit.io/manual/pipeline/filters/lua) give users extreme flexibility with how they want to transform their data, including modifying, adding, enriching via API, calculating, or even redacting. To help us connect Fluent Bit and the new simple schema, we opted for a Lua script that you can paste into your config file to run on top of NGINX ingested logs.
+
+fluent-bit.conf example for VM or standalone deployments:
+
+```bash
+[INPUT]
+    name tail
+    read_from_head true
+    exit_on_eof true
+    path /tmp/data.log
+    parser nginx
+
+[Filter]
+    Name    lua
+    Match   *
+    code    function cb_filter(a,b,c)local d={}local e=os.date("!%Y-%m-%dT%H:%M:%S.000Z")d["observerTime"]=e;d["body"]=c.remote.." "..c.host.." "..c.user.." ["..os.date("%d/%b/%Y:%H:%M:%S %z").."] \""..c.method.." "..c.path.." HTTP/1.1\" "..c.code.." "..c.size.." \""..c.referer.."\" \""..c.agent.."\""d["trace_id"]="102981ABCD2901"d["span_id"]="abcdef1010"d["attributes"]={}d["attributes"]["data_stream"]={}d["attributes"]["data_stream"]["dataset"]="nginx.access"d["attributes"]["data_stream"]["namespace"]="production"d["attributes"]["data_stream"]["type"]="logs"d["event"]={}d["event"]["category"]={"web"}d["event"]["name"]="access"d["event"]["domain"]="nginx.access"d["event"]["kind"]="event"d["event"]["result"]="success"d["event"]["type"]={"access"}d["http"]={}d["http"]["request"]={}d["http"]["request"]["method"]=c.method;d["http"]["response"]={}d["http"]["response"]["bytes"]=tonumber(c.size)d["http"]["response"]["status_code"]=c.code;d["http"]["flavor"]="1.1"d["http"]["url"]=c.path;d["communication"]={}d["communication"]["source"]={}d["communication"]["source"]["address"]="127.0.0.1"d["communication"]["source"]["ip"]=c.remote;return 1,b,d end
+    call    cb_filter
+
+[OUTPUT]
+    format json
+    name stdout
+    match *
+```
+
+## Importing the dashboard in OpenSearch
+
+See the following tutorials:
```

Comment:
This sentence needs more information. Why should the reader see the tutorials? What will the tutorials help them do?

Reply:
@ariamarble It looks like this comment hasn't been addressed yet.

Reply:
yeah, seem to have missed it. got my change done.